lib/text_diff.ml

```diff
@@ -54,40 +54,138 @@ let pp_ diff_printer =
   in
   Fmt.list ~sep:Fmt.nop pp_dh
 
+let indent amount str =
+  let indentation = String.make amount ' ' in
+  indentation ^ str
+
+let indent_hunk amount hunk =
+  match hunk with
+  | Common lst -> Common (List.map (indent amount) lst)
+  | Line_conflict { orig; new_ } ->
+      Line_conflict
+        {
+          orig = List.map (indent amount) orig;
+          new_ = List.map (indent amount) new_;
+        }
+  | Inline_hunks ihunks ->
+      Inline_hunks
+        (match ihunks with
+        | [] -> ihunks
+        | ihunk :: ihunks' ->
+            (match ihunk with
+            | Icommon s -> Icommon (indent amount s)
+            | Iconflict { iorig; inew } ->
+                Iconflict
+                  {
+                    iorig = Option.map (indent amount) iorig;
```

Comment:
That would store the extra spaces in the conflict and we don't want that. You can simply append an `Icommon s` where `s` is the indentation.

Another solution would be to store the indentation level directly into the hunk so that this function does not have to allocate new strings and this can be dealt with when actually displaying the diff.

Reply:
> Another solution would be to store the indentation level directly into the hunk so that this function does not have to allocate new strings and this can be dealt with when actually displaying the diff.

I totally agree, but I think we should take of refining the textual diff types to take care of spacing and indentation in separate PR, and maybe defining our own pretty printers for our IR of type declarations.